openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -566,25 +595,25 @@ def __setattr__(self, name, value):
                 if not hasattr(request, name):
                     original_user = value
                     if hasattr(value, 'id'):
-                        log.info(
+                        self.user_changes.append(
```

Comment:
Note: I think we might be able to reduce processing overhead by making these functions that delay processing, but I do not think it is worth it. I think it would be better to just leave the toggle off when we aren't collecting data.